packages/components/src/color-picker/use-deprecated-props.ts

```diff
@@ -62,51 +70,32 @@ export interface LegacyProps {
 function isLegacyProps( props: any ): props is LegacyProps {
 	return (
 		typeof props.onChangeComplete !== 'undefined' ||
-		typeof props.color === 'string' ||
+		typeof props.disableAlpha !== 'undefined' ||
 		typeof props.color?.hex === 'string'
 	);
 }
 
-function getColorFromLegacyProps(
-	props: LegacyProps
-): ColorFormats.HSL | ColorFormats.HSLA | undefined {
-	if ( ! props.color ) {
+function getColorFromLegacyProps( props: LegacyProps ): string | undefined {
+	if ( props.color === undefined ) {
```

Comment:
Nit, but this way of checkin may be slightly more robust:

```suggestion
	if ( typeof props?.color === 'undefined' ) {
```